src/vegur.erl

```diff
@@ -138,31 +156,58 @@ defaults() ->
-    {RequestIdMaxSize, Config7} = get_default(request_id_max_size, Config6,
-                                              vegur_utils:config(request_id_max_size)),
-    {DownstreamConnectTimeout, Config8} = get_default(downstream_connect_timeout, Config7,
-                                                      vegur_utils:config(downstream_connect_timeout)),
-    ok = set_config(middleware_stack, Middlewares),
+    { Acceptors,
+      MaxConnections,
+      NewConfig } = prestart_config(Config),
     ok = set_config(interface_module, Interface),
-    ok = set_config(request_id_name, RequestIdName),
-    ok = set_config(connect_time_header, ConnectTime),
-    ok = set_config(route_time_header, RouteTimeHeader),
-    ok = set_config(request_id_max_size, RequestIdMaxSize),
-    ok = set_config(downstream_connect_timeout, DownstreamConnectTimeout),
-    start_listener(Type, Ref, Port, Acceptors, MaxConnections, Config8).
+    start_listener(Type, Ref, Port, Acceptors, MaxConnections, NewConfig).
+
+-spec prestart_config(Config :: proplists:proplists()) ->
+                             { Acceptors :: pos_integer(),
+                               MaxConnections :: pos_integer(),
+                               NewConfig :: proplists:proplists() }.
+prestart_config(Config) ->
+    lists:foldl(
+      fun({ConfigName, AppConfigName, Default},
+          {Acceptors, MaxConnections, ConfigIn}) ->
+              {Value, ConfigOut} = get_default(ConfigName, ConfigIn, Default),
+              ok = set_config(AppConfigName, Value),
+              {Acceptors, MaxConnections, ConfigOut};
+         (acceptors, {_, MaxConnections, ConfigIn}) ->
+              {Acceptors, ConfigOut} =
+                  get_default(acceptors,
+                              ConfigIn,
+                              vegur_utils:config(acceptors)),
+              {Acceptors, MaxConnections, ConfigOut};
+         (max_connections, {Acceptors, _, ConfigIn}) ->
+              {MaxConnections, ConfigOut} =
+                  get_default(max_connections,
+                              ConfigIn,
+                              vegur_utils:config(max_connections)),
+              {Acceptors, MaxConnections, ConfigOut}
+      end,
+      %% Defaults
+      {vegur_utils:config(acceptors), vegur_utils:config(max_connections), Config},
+      [
+       {middlewares, middleware_stack, default_middlewares()},
+       {request_id_header, request_id_name,
+        vegur_utils:config(request_id_name)},
+       {connect_time_header, connect_time_header,
+        vegur_utils:config(connect_time_header)},
+       {route_time_header, route_time_header,
+        vegur_utils:config(route_time_header)},
+       {request_id_max_size, request_id_max_size,
+        vegur_utils:config(request_id_max_size)},
+       {downstream_connect_timeout,
+        downstream_connect_timeout,
+        vegur_utils:config(downstream_connect_timeout)},
+       acceptors,
+       max_connections
+      ]).
```

Comment:
snit uses it because some values are to override default ones -- impossible to change them. I don't think this is required to the same extent here where we allow the user to override everything?